.github/workflows/run_periodic_tests.yml

```diff
@@ -125,6 +125,8 @@ jobs:
 
   run_example_tests:
     runs-on: ubuntu-latest
+    env:
+      BUILD_IDAKLU: ON
```

Comment:
Can we control the idaklu build with something besides environment variables?

Reply:
`scikit-build-core` provides support for [overrides](https://scikit-build-core.readthedocs.io/en/latest/overrides.html#any-matching-condition) based on environment variables. Until new features are added upstream, we have to deal with environment variables. Apart from this, we can have fallback options if the builds fail, but yes, they need to run first.

Reply:
Having to export does not seem like a great interface. I would prefer that it defaults to on and tries to recover if something is not found

Reply:
The reason I am saying this is that it would probably be good to move towards idaklu being the default solver

Reply:
I see. I know that Windows tests and the Ubuntu doctests were failing because we haven't made compiling the IDAKLU solver optional and turned it off.
We can default it to on and track an environment variable to turn it off wherever it isn't needed. Would you like me to work on it?

Reply:
One thing I noticed is that, during CI builds, even if `scikit-build-core` fallbacks to pure python build if the `IDAKLU` compilation fails, it still fails of `AssertionError: Metadata mismatch in WHEEL`. So, I don't think we can set a fall back, even for local development..? Please let me know if I am wrong

Reply:
I tried running tests and building docs. They didn't fail and worked fine. It is because I have all the libs beforehand (also I use a `conda` shell, but the `nox` session builds worked fine). It would cause problems for people newly installing PyBaMM and others along the same line.

Reply:
Just recreated a build failure locally, and yes, the fallback to pure python installation fails because of `AssertionError: Metadata mismatch in WHEEL`